thor/request.py

```diff
@@ -49,39 +75,63 @@ def handleRequest(arguments, ncFileDictTree, log):
-                        return climateAreaDict
-
-                    # Interpolating the data
-                    returnAreaDict = sigProcess.interpolate(
-                        climateAreaDict["data"],
-                        arguments["return-dimension"])
-
-                    if not returnAreaDict["ok"]:
-                        return returnAreaDict
-
-                    return {"ok": True,
-                            "data": returnAreaDict["data"]}
-
-    returnDataDict = {"ok": False,
-                      "errorMessage":
-                      "Specified date range not within server dataset."}
-
-    return returnDataDict
+           and arguments["to-date"] < ncFile.getLastDate():
+            climateAreaDict = ncFile.getData(
+                arguments["from-date"],
+                arguments["to-date"],
+                arguments["from-latitude"],
+                arguments["to-latitude"],
+                arguments["from-longitude"],
+                arguments["to-longitude"])
+
+            if climateAreaDict["ok"]:
+                [latScale,
+                 lonScale] = overlapScale(climateAreaDict["area"],
+                                          arguments["from-latitude"],
+                                          arguments["to-latitude"],
+                                          arguments["from-longitude"],
+                                          arguments["to-longitude"])
+
+                latInterpolLen = floor(arguments["return-dimension"][0] *
+                                       latScale)
+                lonInterpolLen = floor(arguments["return-dimension"][1] *
+                                       lonScale)
+
+                # Interpolating the data
+                returnAreaDict = sigProcess.interpolate(
+                    climateAreaDict["data"],
+                    [1,
+                     latInterpolLen,
+                     lonInterpolLen])
+
+                if returnAreaDict["ok"]:
+
+                    latStart = floor((((climateAreaDict["area"][0] -
+                                        arguments["from-latitude"]) *
+                                       arguments["return-dimension"][0]) /
+                                      (arguments["to-latitude"] -
+                                       arguments["from-latitude"])))
+
+                    lonStart = floor((((climateAreaDict["area"][2] -
+                                        arguments["from-longitude"]) *
+                                       arguments["return-dimension"][1]) /
+                                      (arguments["to-longitude"] -
+                                       arguments["from-longitude"])))
+
+                    returnData[latStart:latStart +
+                               latInterpolLen,
+                               lonStart:lonStart +
+                               lonInterpolLen] = returnAreaDict["data"]
+
+    return {"ok": True,
+            "data": returnData}
```

Comment:
Possibility to return empty data. Check for this and return error if no data is to be returned from any file.